farmbot_core/lib/farmbot_core/asset/command.ex

```diff
@@ -6,17 +6,36 @@ defmodule FarmbotCore.Asset.Command do
   alias FarmbotCore.{
     Asset,
     Asset.Repo,
-    Asset.Device
+    Asset.Device,
+    Asset.FbosConfig
   }
 
-  @type kind :: Device | FbosConfig | FwConfig
-  @type return_type ::
-          Device.t() | FbosConfig.t() | FwConfig.t()
+  @type kind :: Device | FbosConfig | FwConfig | FarmwareEnv
+  @type params :: map()
+  @type id :: float()
+  @type return_type :: Device.t() | FbosConfig.t() | FwConfig.t()
 
-  @callback update(kind, params :: map()) :: return_type
+  @callback update(kind, params) :: return_type
```

Comment:
you can change this to be
```elixir
@callback update(arg1, map) :: Device.t() when arg1 :: Device.t()
```
etc

if it is left as is, dialyzer will throw an error that says anything using this function needs to check for _every_ `kind`. That said, we aren't using Dialyzer right now so it doesn't really matter. 

It's also usually frowned upon to do pattern matches like`update(Device, params)` and is usually favored
to do multiple functions:

```elixir
def update_device(params)
def update_fw_config(params)
def update_farmware_env(params)
```
etc

however, I don't really care personally which way we go